### ISSUE TITLE
By default, what is the best model?

### ISSUE THREAD

**Issue by CatarinaPC** [flairNLP/flair#1472]
By default what is the best model, when training a sequence tagger model? It is the one with lowest training loss or highest f1-score on the dev set?

**Comment by djstrong** [comment]
Highest f1-score on the dev set.

**Comment by CatarinaPC** [comment]
#1 - And the training stops (until a maximum of a certain number of epochs) when the learning rate gets to a certain low value?

#2 - And the learning rate decreases at each epoch if the dev score does not improve (= bad epoch?) ?

#3 - after a certain amount of bad epochs does the training stop ? or does it stop when the learning rate gets to small?

#4 - is this what annealing is?

**Comment by djstrong** [comment]
After `patience` epochs without improvement the learning rate is decreased by `anneal_factor`. The training stops after `max_epochs` or when learning rate is smaller than `min_learning_rate`.

**Comment by alanakbik** [comment]
Closing issues since question is answered (thanks @djstrong!) but feel free to reopen if there are further questions!
